public/app/plugins/panel/geomap/module.tsx

```diff
@@ -32,14 +38,62 @@ export const plugin = new PanelPlugin<GeomapPanelOptions>(GeomapPanel)
       defaultValue: defaultView.shared,
     });
 
-    builder.addCustomEditor({
-      category: ['Base Layer'],
-      id: 'basemap',
-      path: 'basemap',
-      name: 'Base Layer',
-      editor: BaseLayerEditor,
-      defaultValue: DEFAULT_BASEMAP_CONFIG,
-    });
+    if (config.geomapDisableCustomBaseLayer) {
+      builder.addCustomEditor({
+        category: ['Base Layer'],
+        id: 'basemap',
+        path: 'basemap',
+        name: 'Base Layer',
+        // eslint-disable-next-line react/display-name
+        editor: () => {
+          return <div>The base layer is configured by the server admin.</div>;
+        },
+        defaultValue: DEFAULT_BASEMAP_CONFIG,
+      });
+    } else {
+      // Append the basemap prefix
+      builder.setDefaultConfig({
+        category: ['Base Layer'],
+        beforeChange: ({ path, value }) => {
+          return { path: `basemap.${path}`, value };
+        },
+        valueGetter: (root: any, path: string) => {
+          return lodashGet(root, `basemap.${path}`);
+        },
```

Comment:
Wonder if we need these functions at all? Wouldn't 'rootPath: string` be enough? I'm especially worried about the `beforeChange` as it only applies the value to a correct path. With the `rootPath` configured for the consecutive options this could be done under the hood.